### PR TITLE
Add OptionButton control

### PR DESCRIPTION
**Pull request by promag**
image

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/Win64%20\[unit%20tests,%20no%20gui%20tests,%20no%20boost::process,%20no%20functional%20tests\]%20\[focal\]/insecure_win_gui.zip?branch=pull/67)
[macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macOS%2010.15%20\[gui,%20no%20tests\]%20\[focal\]/insecure_mac_gui.zip?branch=pull/67)

### CODE REVIEW

**Review by hebasto — COMMENTED** [review]
Tested 7cde50b6a144ad87f6d05dfa100f9ecb29974b86 on Linux Mint 20.2 (Qt 5.12.8).

`Tab` (focus moving) and `Space` (focus selecting) work as expected.

What should be a role of the `Return` key in focus treatment?

Also:
```
$ src/qt/bitcoin-qt -signet
qrc:/qml/pages/stub.qml:34:9: QML Pane: Binding loop detected for property "implicitHeight"
qrc:/qml/pages/stub.qml:34:9: QML Pane: Binding loop detected for property "implicitHeight"
```

**Comment by promag** [comment]
Thanks @hebasto, will address both comments.

File: src/qml/controls/OptionButton.qml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    id: button
+    padding: 15
+    checkable: true
+    background: Rectangle {
+        border.width: 1
+        border.color: button.checked ? "#F7931A" : button.hovered ? "white" : "#999999"
+        radius: 10
+        color: "transparent"
+        Rectangle {
+            visible: button.activeFocus
```
Comment (promag) [inline src/qml/controls/OptionButton.qml:21]: Will use `visualFocus` instead, see https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html#visualFocus-prop.

**Comment by promag** [comment]
Rebased and fixed warning. Now using `visualFocus` instead of `activeFocus`. Clicking or touching doesn't show focus border, only if the keyboard is used.

File: src/qml/bitcoin_qml.qrc [hunk 2]
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE RCC><RCC version="1.0">
     <qresource prefix="/qml">
+        <file>controls/OptionButton.qml</file>
```
Comment (hebasto) [inline src/qml/bitcoin_qml.qrc:3]: dc30b7085be0fa58e568515c730d2a7365c32d63

Sort lexicographically?

**Review by hebasto — APPROVED** [review]
ACK 8ab1626be7e717c6172582a89f70346481166778

File: src/qml/bitcoin_qml.qrc (hunk 2 above)
Comment (promag) [reply src/qml/bitcoin_qml.qrc]: Done.

**Review by hebasto — APPROVED** [review]
re-ACK 61668e989adc3fc53431665fddf4fd1f19926391

File: src/qml/pages/stub.qml [hunk 3]
```diff
@@ -5,33 +5,35 @@
...
 import QtQuick.Layouts 1.11
-import "../components" as BitcoinCoreComponents
-
+import "../components"
 
 ApplicationWindow {
     id: appWindow
     title: "Bitcoin Core TnG"
     minimumWidth: 750
     minimumHeight: 450
```
Comment (jarolrod) [inline src/qml/pages/stub.qml:14]: Do we have any plan on how to set the `minimumHeight` to correspond to the required minimum size of the contents?
Screen Shot 2021-11-02 at 7 27 24 PM

**Review by jarolrod — COMMENTED** [review]
ACK 61668e9

pending on ACK from our fellow designers. Pinging @Bosch-0 @GBKS

File: src/qml/pages/stub.qml (hunk 3 above)
Comment (promag) [reply src/qml/pages/stub.qml:14]: We should set these once we settle down the main/window layout.

**Comment by Bosch-0** [comment]
ACK, LGTM on windows, this will be siloed to the onboarding / settings sections at a later stage correct?

**Pull request merged**